libpod/networking_linux.go

```diff
@@ -1047,21 +1047,24 @@ func (c *Container) NetworkDisconnect(nameOrID, netName string, force bool) erro
 		return err
 	}
 
-	if c.state.State != define.ContainerStateRunning {
-		return errors.Wrapf(define.ErrCtrStateInvalid, "cannot disconnect container %s from networks as it is not running", nameOrID)
+	if err := c.runtime.state.NetworkDisconnect(c, netName); err != nil {
+		return err
 	}
-	if c.state.NetNS == nil {
-		return errors.Wrapf(define.ErrNoNetwork, "unable to disconnect %s from %s", nameOrID, netName)
+
+	c.newNetworkEvent(events.NetworkDisconnect, netName)
+	if c.state.State != define.ContainerStateRunning {
+		return nil
 	}
+
 	podConfig := c.runtime.getPodNetwork(c.ID(), c.Name(), c.state.NetNS.Path(), []string{netName}, c.config.PortMappings, nil, nil, c.state.NetInterfaceDescriptions)
 	if err := c.runtime.netPlugin.TearDownPod(podConfig); err != nil {
 		return err
 	}
-	if err := c.runtime.state.NetworkDisconnect(c, netName); err != nil {
-		return err
-	}
 
-	// update network status
+	if c.state.NetNS == nil {
```

Comment:
I think this has to be above l1059 to prevent a potential null dereference panic when getting `c.state.NetNS.Path()`